### ISSUE TITLE
Tensorpack trainer doesn't update `non_trainable_variables`

### ISSUE THREAD

**Issue by see--** [tensorpack/tensorpack#1202]
I added one `BatchNormalization` layer to `tensorpack/examples/keras/mnist-keras.py`. It looks like `non_trainable_variables` (i.e. `moving_mean`, `moving_variance`) don't get updates? It is no problem with the `KerasModel` interface.

If you're asking about an unexpected problem which you do not know the root cause,
use this template. __PLEASE DO NOT DELETE THIS TEMPLATE, FILL IT__:

If you already know the root cause to your problem,
feel free to delete everything in this template.

### 1. What you did:

(1) **If you're using examples, what's the command you run:**
`python examples/keras/mnist-keras.py`
After training:
`python scripts/ls-checkpoint.py train_log/mnist-keras/model-9828`

This gives:
```
 'batch_normalization_v1/moving_mean': 0.0,
 'batch_normalization_v1/moving_variance': 1.0
```

(2) **If you're using examples, have you made any changes to the examples? Paste `git status; git diff` here:**
```
diff --git a/examples/keras/mnist-keras.py b/examples/keras/mnist-keras.py
index 4e34b9d5..8796f51f 100755
--- a/examples/keras/mnist-keras.py
+++ b/examples/keras/mnist-keras.py
@@ -42,6 +42,7 @@ def get_keras_model():
     with clear_tower0_name_scope():
         M = keras.models.Sequential()
         M.add(KL.Conv2D(32, 3, activation='relu', input_shape=[IMAGE_SIZE, IMAGE_SIZE, 1], padding='same'))
+        M.add(KL.BatchNormalization())
         M.add(KL.MaxPooling2D())
         M.add(KL.Conv2D(32, 3, activation='relu', padding='same'))
         M.add(KL.Conv2D(32, 3, activation='relu', padding='same'))
diff --git a/scripts/ls-checkpoint.py b/scripts/ls-checkpoint.py
index cb4cabed..1f5c36d4 100755
--- a/scripts/ls-checkpoint.py
+++ b/scripts/ls-checkpoint.py
@@ -23,4 +23,7 @@ if __name__ == '__main__':
         path = get_checkpoint_path(fpath)
         reader = tf.train.NewCheckpointReader(path)
         dic = reader.get_variable_to_shape_map()
+        dic2 = {k: reader.get_tensor(k).mean() for k in dic}
     pprint.pprint(dic)
+    pprint.pprint(dic2)
+
```

(3) **If not using examples, tell us what you did:**

  It's always better to copy-paste what you did than to describe them.

  Please try to provide enough information to let other __reproduce__ your issues. 
  Without reproducing the issue, we may not be able to investigate it.

### 2. What you observed:

(1) **Include the ENTIRE logs here:**

It's always better to copy-paste what you observed instead of describing them.

It's always better to paste **as much as possible**, although sometimes a partial log is OK.

Tensorpack typically saves stdout to its training log.
If stderr is relevant, you can run a command with `CMD 2>&1 | tee logs.txt`
to save both stdout and stderr to one file.

(2) **Other observations, if any:**
For example, CPU/GPU utilization, output images, tensorboard curves, if relevant to your issue.

### 3. What you expected, if not obvious.
```
 'batch_normalization_v1/moving_mean' != 0.0,
 'batch_normalization_v1/moving_variance' != 1.0
```

If you expect higher speed, please read
http://tensorpack.readthedocs.io/tutorial/performance-tuning.html 
before posting.

If you expect certain accuracy, only in one of the two conditions can we help with it:
(1) You're unable to reproduce the accuracy documented in tensorpack examples.
(2) It appears to be a tensorpack bug.

Otherwise, how to train a model to certain accuracy is a machine learning question.
We do not answer machine learning questions and it is your responsibility to 
figure out how to make your models more accurate.

### 4. Your environment:
  + Paste the output of this command: `python3 -c 'import tensorpack.tfutils as u; print(u.collect_env_info())'`
    If this command failed, tell us your version of Python/TF/tensorpack.
  + You can install Tensorpack master by `pip install -U git+https://github.com/ppwwyyxx/tensorpack.git`
    and see if your issue is already solved.
  + If you're not using tensorpack under a normal command line shell (e.g.,
    using an IDE or jupyter notebook), please retry under a normal command line shell. 
  + Include relevant hardware information, e.g. number of GPUs used for training, amount of RAM.
```
--------------------  -------------------------------------------------------------------
sys.platform          linux
Python                3.6.7 (default, Oct 22 2018, 11:32:17) [GCC 8.2.0]
Tensorpack            0.9.4
Numpy                 1.16.3
TensorFlow            1.13.1/b'v1.13.1-0-g6612da8951'
TF Compiler Version   4.8.5
TF CUDA support       True
TF MKL support        False
Nvidia Driver         /usr/lib/x86_64-linux-gnu/libnvidia-ml.so.418.67
CUDA                  /usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudart.so.10.0.130
CUDNN                 /usr/lib/x86_64-linux-gnu/libcudnn.so.7.5.1
NCCL
CUDA_VISIBLE_DEVICES  None
GPU 0                 GeForce GTX 1080 Ti
Free RAM              14.21/15.61 GB
CPU Count             12
cv2                   3.4.3
msgpack               0.6.1
python-prctl          True
--------------------  -------------------------------------------------------------------
```
You may often want to provide extra information related to your issue, but
at the minimum please try to provide the above information __accurately__ to save effort in the investigation.

**Comment by ppwwyyxx** [comment]
Keras updates batch norm by `model.updates`.
Therefore you need to add ops in `M.updates` into the collection `tf.GraphKeys.UPDATE_OPS` so that they can be known by tensorpack, like this:
https://github.com/tensorpack/tensorpack/blob/2981c5d4336ff36c158aa0fb5294975c80311c9c/tensorpack/contrib/keras.py#L75-L76

**Comment by ppwwyyxx** [comment]
This should be fixed in latest master now. Please note again that Keras support is experimental and I don't think there is a point for tensorpack to fully support Keras models except to demo a proof-of-concept of such possibility.

**Comment by see--** [comment]
Thanks a lot! BTW I think it would be great to have full support for option 1:
> Write the tower function similar to a standard tensorpack program, but use some Keras layers in between. See mnist-keras.py on how to do this. It does not support all tensorpack trainers.

I am not really familiar with the keras variable scope issue. Is it something that can be fixed in the TensorFlow repo?

**Comment by ppwwyyxx** [comment]
It is possible to fix but it won't be fixed by them, because supporting variable scope is not useful in Keras.
